### ISSUE TITLE
sourcesContent option

### ISSUE THREAD

**Issue by guybedford** [systemjs/builder#68]
Support inlining all sources into the source map itself.

**Comment by guybedford** [comment]
(Not a high priority currently).

**Comment by hgl** [comment]
Thanks for creating the issue.

Maybe could also provide an option to inline the source map itself in the bundled file with data url?

**Comment by twalker** [comment]
+1

**Comment by crisptrutski** [comment]
@guybedford cost here is low, intermediate source maps (via traceur / 6to5) actually contain `sourcesContent`, we just need to preserve them across concatenation - have got that working, and will put it behind a flag.

There's one gotcha - transitive source maps have the wrong content (after plugin compilation). Anxious this might be a bug upstream, confirming now.

**Comment by crisptrutski** [comment]
OK, the chaining issue is actually related in this case to the source map produced by `react-tools` (no `sourcesContent`). 

In my testing both 6to5 and Traceur are returning source maps listing the jsx file as the source, but with the compiled source content, so this seems to be a bug back in the mozilla library. Will PR back there, we can handle gaps but not lies.

For gaps, I guess we'll just dig back to the file system (to ensure misc plugins can't ruin the parade)

**Comment by guybedford** [comment]
@crisptrutski transitive sourcesContent bubbling up correctly would be ideal and fills gaps as you say. Happy to see this as a nice-to-have feature though, just like the issue here.

**Comment by crisptrutski** [comment]
@twalker as you mentioned on the other thread this being a big deal for you, pushed up a crude solution here https://github.com/systemjs/builder/tree/wip-sources-content if that's useful to you in the mean time.

**Comment by guybedford** [comment]
:+1:

**Comment by twalker** [comment]
Thank you very much @crisptrutski . 

My goal is to transpile with 6to5 and be able to debug ES6 source files. Inlining the source seems less prone to pathing errors.

I'm currently blocked by https://github.com/jspm/jspm-cli/issues/511 so I think I need to roll back instead of forward. I fully appreciate the effort going into getting sourcemaps working though.

**Comment by guybedford** [comment]
Yeah this would definitely be useful for source map portability.

**Comment by crisptrutski** [comment]
Next on my list after auto run-time injection. Likely this weekend.

**Comment by guybedford** [comment]
@crisptrutski can you provide a code example of how I would create a separate source map that inlines all the sources it references?

**Comment by crisptrutski** [comment]
@guybedford you mean in README? If you just want to see code, https://github.com/systemjs/builder/blob/master/test/sourcemaps.js#L133-L151

**Comment by crisptrutski** [comment]
(emoji links work, awesome)

**Comment by guybedford** [comment]
Nice! Yes it would be great to have a code snippet in the readme.

**Comment by crisptrutski** [comment]
Do you want that immediately? Thought README only updates on release (or is that only for jspm?)

**Comment by guybedford** [comment]
We did release this already?

**Comment by crisptrutski** [comment]
No, came in after `0.10.4`

**Comment by guybedford** [comment]
Ahh right, sorry my mistake.

**Comment by crisptrutski** [comment]
Put fixed-pending-release tag back back, put a sticky in checklist to add to README?

**Comment by guybedford** [comment]
Thanks, that would be great.

**Comment by OliverJAsh** [comment]
Is there any documentation for this feature?

I want to add `sourcesContent` to my source map, but I still want the source map to be a separate file.

**Comment by OliverJAsh** [comment]
Got it: `sourcesContent: true`
